lib/chef/provider/osx_profile.rb

```diff
@@ -0,0 +1,258 @@
+              /^\w+(?:\.\w+)+$/.match(new_profile_name)
+            }
+            a.failure_message RuntimeError, "When removing by resource name, it must match the profile identifier "
+          end
+        end
+
+        requirements.assert(:install) do |a|
+          if @new_profile_hash.is_a?(Hash)
+            a.assertion {
+              @new_profile_hash.include?('PayloadIdentifier')
+            }
+            a.failure_message RuntimeError, "The specified profile does not seem to be valid"
+          end
+        end
+      end
+
+      def action_install
+        unless @current_resource.installed
+          converge_by("install profile #{@new_profile_identifier}") do
+            profile_path = write_profile_to_disk
+            install_profile(profile_path)
+            get_installed_profiles(update=true)
+          end
+        end
+      end
+
+      def action_remove
+        # Clean up profile after removing it
+        if @current_resource.installed
+          converge_by("remove profile #{@new_profile_identifier}") do
+            remove_profile
+            get_installed_profiles(update=true)
+          end
+        end
+      end
+
+      def load_profile_hash(new_profile)
+          # file must exist in cookbook
+          unless new_profile.end_with?('.mobileconfig')
+            error_string = "'#{new_profile}' is not a valid profile"
+            raise RuntimeError, error_string
+          end
+          unless cookbook_file_available?(new_profile)
+            error_string = "#{self.to_s}: '#{new_profile}' not found in cookbook"
+            raise Chef::Exceptions::FileNotFound, error_string
+          end
+          # load the profile
+          cookbook_profile = cache_cookbook_profile(new_profile)
+          profile = read_plist(cookbook_profile)
+          profile
+      end
+
+      def cookbook_file_available?(cookbook_file)
+        run_context.has_cookbook_file_in_cookbook?(
+          @new_resource.cookbook_name, cookbook_file
+        )
+      end
+
+      def get_cache_dir
+        cache_dir = Chef::FileCache.create_cache_path(
+          "profiles/#{@new_resource.cookbook_name}"
+        )
+      end
+
+      def cache_cookbook_profile(cookbook_file)
+        Chef::FileCache.create_cache_path(
+          ::File.join(
+            "profiles",
+            @new_resource.cookbook_name,
+            ::File.dirname(cookbook_file)
+          )
+        )
+        remote_file = Chef::Resource::CookbookFile.new(
+          ::File.join(
+            get_cache_dir,
+            "#{cookbook_file}.remote"
+          ),
+          run_context
+        )
+        remote_file.cookbook_name = @new_resource.cookbook_name
+        remote_file.source(cookbook_file)
+        remote_file.backup(false)
+        remote_file.run_action(:create)
+        remote_file.path
+      end
+
+      def get_profile_hash(new_profile)
+        if @new_resource.action.include?(:install)
```

Comment:
I see the logic here, but in the case of a removal, if the profile_name is set to the identifier, it should use that as the profile_identifier.

This allows a smaller resource call as such:
``` 
 profile 'com.cpe.profiles.screensaver' do
    action :remove
  end
```

Otherwise, you would need to specify:
```
  profile 'Remove some profile' do
    identifier 'com.cpe.profiles.screensaver'
    action :remove
  end
```

If it is a removal, it should know the resource name should be used as the identifier instead of profile name. 

One solution is if it ends in .mobileconfig, assume it is an install, if it does not, assume it is an identifier? This also seems kind of janky. I am assuming nothing with the current code, but if I have a better way to do it in a predictable fashion, I am happy to do it. (Perhaps I am missing something?)